packages/git/src/browser/blame/blame-contribution.ts

```diff
@@ -0,0 +1,130 @@
+import { CommandContribution, CommandRegistry, Command, MenuContribution, MenuModelRegistry, Disposable, DisposableCollection } from '@theia/core/lib/common';
+import { BlameDecorator } from './blame-decorator';
+import { EDITOR_CONTEXT_MENU, EditorManager, EditorKeybindingContexts, EditorWidget } from '@theia/editor/lib/browser';
+import { BlameManager } from './blame-manager';
+import URI from '@theia/core/lib/common/uri';
+
+export namespace BlameCommands {
+    export const SHOW_GIT_ANNOTATIONS: Command = {
+        id: 'git.editor.show.annotations',
+        label: 'Show Git Annotations'
+    };
+    export const CLEAR_GIT_ANNOTATIONS: Command = {
+        id: 'git.editor.clear.annotations'
+    };
+}
+
+@injectable()
+export class BlameContribution implements FrontendApplicationContribution, CommandContribution, KeybindingContribution, MenuContribution {
+
+    @inject(EditorManager)
+    protected readonly editorManager: EditorManager;
+
+    @inject(BlameDecorator)
+    protected readonly decorator: BlameDecorator;
+
+    @inject(BlameManager)
+    protected readonly blameManager: BlameManager;
+
+    onStart(app: FrontendApplication): void {
+    }
+
+    registerCommands(commands: CommandRegistry): void {
+        commands.registerCommand(BlameCommands.SHOW_GIT_ANNOTATIONS, {
+            execute: () => {
+                const editor = this.currentEditor;
+                if (editor) {
+                    this.showBlame(editor);
+                }
+            },
+            isVisible: () =>
+                !!this.currentEditor,
+            isEnabled: () => {
+                const editor = this.currentEditor;
+                return !!editor && this.isBlameable(editor.editor.uri);
+            }
+        });
+        commands.registerCommand(BlameCommands.CLEAR_GIT_ANNOTATIONS, {
+            execute: () => {
+                const editor = this.currentEditor;
+                if (editor) {
+                    this.clearBlame(editor.editor.uri);
+                }
+            },
+            isVisible: () =>
+                !!this.currentEditor,
+            isEnabled: () => {
+                const editor = this.currentEditor;
+                return !!editor && this.appliedDecorations.has(editor.editor.uri.toString());
+            }
+        });
+    }
+
+    protected get currentEditor(): EditorWidget | undefined {
+        const editor = this.editorManager.currentEditor;
+        return editor;
+    }
+
+    protected isBlameable(uri: string | URI): boolean {
+        return this.blameManager.isBlameable(uri.toString());
+    }
+
+    protected appliedDecorations = new Map<string, Disposable>();
+
+    protected async showBlame(editorWidget: EditorWidget) {
+        const uri = editorWidget.editor.uri;
+        if (editorWidget.editor.document.dirty) {
+            await editorWidget.editor.document.save();
```

Comment:
Won't it be surprising?

Reply:
I was thinking of several options:
 - save before acquire `git blame`
 - not showing for dirty editors
 - acquire `git blame` for saved version and merge with changes of dirty editor

WDYT which is the better solution?

Reply:
What does VS Code do?

Reply:
<img width="996" alt="screen shot 2018-03-07 at 11 50 11" src="https://user-images.githubusercontent.com/914497/37088465-ba5fc53a-21fd-11e8-9988-ea1db0f51726.png">

third option ;-)

Reply:
gitlense, to be more exact.

Reply:
you cannot annotate removed lines.

new lines are only added to the "uncommitted" range, otherwise once you modify some annotated line, the annotations are removed. 

current state of this PR: each modification removes the annotations.

Reply:
ok, checked with Eclipse, they seem to do the same.

Reply:
👍 thanks for checking!